tests/test_audio.py

```diff
@@ -95,3 +100,73 @@ def test_match_sample_length():
             pytest.raises(ScaperError, match_sample_length, siren.name, _duration)
+
+
+def test_peak_normalize():
+
+    def sine(x, f, sr, A):
+        return A * np.sin(2 * np.pi * f * x / sr)
+
+    def square(x,f, sr, A):
+        return A * sg.square(2 * np.pi * f * x / sr)
+
+    def saw(x, f, sr, A):
+        return A * sg.sawtooth(2 * np.pi * f * x / sr)
+
+    samplerates = [16000, 44100]
+    frequencies = [50, 100, 500, 1000, 5000, 10000, 15000, 20000]
+    amplitudes = [0.1, 0.5, 1.0, 1.5, 2.0]
+    event_factors = [0.5, 0.8]
+    eps = 1e-10
+
+    # test with toy data
+    for waveform in [sine, square, saw]:
+        for sr in samplerates:
+            for f in frequencies:
+                for A in amplitudes:
+
+                    n_samples = sr
+                    x = np.arange(n_samples)
+                    audio = waveform(x, f, sr, A)
+                    max_sample = np.max(np.abs(audio))
+                    estimated_scale_factor = 1.0 / (A + eps)
+                    print("\nsr: {}, f: {}, A: {}".format(sr, f, A))
+                    print("max sample audio: {}".format(max_sample))
+                    print("estimated scale_factor: {}".format(estimated_scale_factor))
+
+                    event_audio_list = []
+                    for factor in event_factors:
+                        event_audio_list.append(waveform(x, f, sr, A * factor))
+
+                    audio_norm, event_audio_list_norm, scale_factor = \
+                        peak_normalize(audio, event_audio_list)
+
+                    print("allclose on factors: {}".format(np.allclose(
+                        scale_factor, estimated_scale_factor)))
+
+                    print("actual scale factor: {}".format(scale_factor))
+
+                    # test scale factor
+                    max_sample_audio = np.max(np.abs(audio_norm))
+                    print("max sample audio norm: {}".format(max_sample_audio))
+                    assert np.allclose(scale_factor, estimated_scale_factor,
+                                       atol=1e-03)
+
+                    # test soundscape audio
+                    assert max_sample_audio < 1.0
+                    assert np.allclose(max_sample_audio, 1.0)
+
+                    # test event audio
+                    for event_audio, factor in zip(event_audio_list_norm,
+                                                   event_factors):
+
+                        max_sample_event = np.max(np.abs(event_audio))
+
+                        if not np.allclose(
+                                max_sample_event, A * factor * scale_factor):
+                            print(max_sample_audio, max_sample_event,
+                                  A * factor * scale_factor)
+
+                        assert np.allclose(max_sample_event,
+                                           A * factor * scale_factor,
+                                           atol=1e-3)
```

Comment:
for some reason GH UI isn't letting me respond to the previous 2 comments, so answering here for all 3:
* low atol because any higher and test fails, I think it's floating point issues
* < 1.0 because we add eps so it should never hit 1